service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java

```diff
@@ -61,29 +65,25 @@ public Function<KStream<String,Data>, KStream<String,Data>> fileProcessor(@Quali
                     Data result = retryTemplate.execute(
                             retryContext -> {
                                 try {
-
                                     fileStatusService.updateEntry(value,Status.IN_PROGRESS);
                                     ClassLoader classLoader = getClass().getClassLoader();
                                     File file = new File(classLoader.getResource(".").getFile() + value.getFileName());
                                     if (file.exists()) {
-                                        fileStatusService.updateEntry(value,Status.FAILURE);
+                                        this.kafkaTemplate.send(TOPIC,fileStatusService.updateEntry(value,Status.FAILURE));
```

Comment:
Why do we need to add code to send to topic? Doesnt Kafka streams take care of it on ts own?

Reply:
Branced stream has been implemented.